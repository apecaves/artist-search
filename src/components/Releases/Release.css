.Release {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
  color: black;
  text-decoration: none;
  &:focus,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
  }
  &:hover {
    color: grey;
  }
}
