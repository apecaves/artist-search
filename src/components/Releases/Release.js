import React from 'react';
import PropTypes from 'prop-types';

function Release({ release }) {
  const imgURL = `http://coverartarchive.org/release/${release.id}/front`;
  return (
    <>
      <p>{release.title}</p>
      <p>{release.date}</p>
      <p>{release.country}</p>
      <img src={imgURL} />
    </>
  );
}

Release.propTypes = {
  release: PropTypes.shape({
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    country: PropTypes.string.isRequired
  }).isRequired
};

export default Release;
