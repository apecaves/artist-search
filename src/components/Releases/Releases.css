.Releases { 
  margin: 0 auto;
  padding: 0;
  list-style: none;
  display: grid; 
  grid-template-columns: 1fr 1fr 1fr 1fr ; 
  grid-template-rows: 1fr 1fr 1fr 1fr 1fr 1fr; 
  grid-column-gap: 10px;
  grid-row-gap: 10px; 

}

.bandName{
  text-align: center;
  border: 1px dotted black;
}