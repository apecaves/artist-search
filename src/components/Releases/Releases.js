import React from 'react';
import PropTypes from 'prop-types';
import Release from './Release';

function Releases({ releaseList }) {
  const releaseItem = releaseList.map(release => {
    return (
      <li key={Release.id}>
        <Release release={release}/>
      </li>
    );
  });

  return <ul>{releaseItem}</ul>;
}

Releases.propTypes = {
  releaseList: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Releases;
