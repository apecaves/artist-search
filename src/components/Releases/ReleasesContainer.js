import React, { Component } from 'react';
import Releases from './Releases';
import { getReleaseList } from '../../services/musicBrainsApi';
import PropTypes from 'prop-types';
import Paging from '../Paging';

export default class ReleasesContainer extends Component {
  static propTypes = {
    match: PropTypes.object.isRequired
  }

  state = {
    releases: [],
    page: 1,
    totalPages: 0
  }

  componentDidMount = () => {
    const { page } = this.state;
    return getReleaseList(this.props.match.params.id, page)
      .then((res) => {
        this.setState({ releases: res.releases, page: 1, totalPages: Math.ceil(res['release-count'] / 10) });
      });
  }

  handlePageChange = operator => {
    const { page } = this.state;
    this.setState({ page: Number.parseInt(page + operator) });
    return getReleaseList(this.props.match.params.id, page + operator)
      .then((res) => {
        this.setState({ releases: res.releases, totalPages: Math.ceil(res['release-count'] / 10) });
      });
  };

  render() {
    const { releases, totalPages, page } = this.state;

    return (
      <>
        <Paging handlePageChange={this.handlePageChange} totalPages={totalPages} page={page}/>
        <Releases releaseList={releases} artistName={this.props.match.params.artistName}/>
      </>
    );
  }

}
