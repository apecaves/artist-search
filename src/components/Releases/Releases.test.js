import React from 'react';
import { shallow } from 'enzyme';
import Releases from './Releases';

describe('Releases component', () => {
  it('renders Releases', () => {
    const wrapper = shallow(<Releases releaseList={[]} artistName={''}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  