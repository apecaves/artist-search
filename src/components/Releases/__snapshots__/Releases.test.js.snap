// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Releases component renders Releases 1`] = `<ul />`;
